tests/regression/test_eigensolver.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import pytest
+from firedrake import *
+
+
+def evals(n, degree=1, mesh=None):
+    '''We base this test on the 1D Poisson problem with Dirichlet boundary
+    conditions, outlined in part 1 of Daniele Boffi's
+    'Finite element approximation of eigenvalue problems' Acta Numerica 2010'''
+    # Create mesh and define function space
+    if mesh is None:
+        mesh = IntervalMesh(n, 0, pi)
+    V = FunctionSpace(mesh, "CG", degree)
+    # Define variational problem
+    u = TrialFunction(V)
+    v = TestFunction(V)
+    a = (inner(grad(u), grad(v))) * dx
+
+    # Create eigenproblem with boundary conditions
+    bc = DirichletBC(V, 0.0, "on_boundary")
+    eigenprob = LinearEigenproblem(a, bcs=bc)
+
+    # Create corresponding eigensolver, looking for n eigenvalues
+    eigensolver = LinearEigensolver(eigenprob, n)
```

Comment:
```suggestion
    eigenprob = LinearEigenproblem(a, bcs=bc, bc_shift=-666.)

    # Create corresponding eigensolver, looking for n eigenvalues
    eigensolver = LinearEigensolver(eigenprob, n, solver_parameters={"eps_largest_real": None})
```
to make this test work consistently, I think. (There are `n + 1` eigenvalues in total, of which two are `bc_shift`).

Reply:
`bc_shift` set to zero should also do the trick and be more in keeping with the mathematical theory.